pytorch_lightning/trainer/connectors/debugging_connector.py

```diff
@@ -31,16 +31,30 @@ def on_init_start(
             overfit_batches,
             fast_dev_run
     ):
+        if not isinstance(fast_dev_run, (bool, int)):
+            raise MisconfigurationException(
+                f'fast_dev_run={fast_dev_run} is not a valid configuration.'
+                ' It should be either a bool or an int >= 0'
+            )
+
+        if isinstance(fast_dev_run, int) and (fast_dev_run < 0):
+            raise MisconfigurationException(
+                f'fast_dev_run={fast_dev_run} is not a'
+                ' valid configuration. It should be >= 0.'
+            )
 
         self.trainer.fast_dev_run = fast_dev_run
-        if self.trainer.fast_dev_run:
-            limit_train_batches = 1
-            limit_val_batches = 1
-            limit_test_batches = 1
+        fast_dev_run = int(fast_dev_run)
+
+        if fast_dev_run:
+            limit_train_batches = fast_dev_run
+            limit_val_batches = fast_dev_run
+            limit_test_batches = fast_dev_run
```

Comment:
Hey @rohitgr7, could you set it there instead
```
# When fast_dev_run > 1, we expect model to logs_every_n_steps, etc..
self.trainer.fast_dev_run = `int(self.trainer.fast_dev_run) == 1`
```
So the rest of the code is cleaner

Reply:
the reason I did it this way is because of this warning by keeping the actual value of `fast_dev_run` passed by user in the trainer.
`rank_zero_warn(f'Skipping batch size scaler since `fast_dev_run={trainer.fast_dev_run}`', UserWarning)`

will update it as you suggested.

Reply:
@tchaton I don't think this is a good idea. if `fast_dev_run > 1`, we can't just set `trainer.fast_dev_run = False`. Isn't this wrong?? let me try something else

Reply:
let me know if new changes are good.

Reply:
fast dev run should not do any logging, agreed.